mvc/views/tag-helpers/intro.rst

```diff
@@ -1,8 +1,281 @@
+If you're familiar with `HTML Helpers <http://stephenwalther.com/archive/2009/03/03/chapter-6-understanding-html-helpers>`__, Tag Helpers reduce the explicit transitions between HTML and C# in Razor views. `Tag Helpers compared to HTML Helpers`_ explains the differences in more detail.
+
+What Tag Helpers provide
+------------------------------------
+
+**An HTML-friendly development experience**
+ For the most part, Razor markup using Tag Helpers looks like standard HTML. Front-end designers conversant with HTML/CSS/JavaScript can edit Razor without learning C# Razor syntax.
+
+**A rich IntelliSense environment for creating HTML and Razor markup**
+ This is in sharp contrast to HTML Helpers, the previous approach to server-side creation of markup in Razor views. `Tag Helpers compared to HTML Helpers`_ explains the differences in more detail. `IntelliSense support for Tag Helpers`_ explains the IntelliSense environment. Even developers experienced with Razor C# syntax are more productive using Tag Helpers than writing C# Razor markup. 
+ 
+**A way to make you more productive and able to produce more robust, reliable, and maintainable code using information only available on the server**
+ For example, historically the mantra on updating images was to change the name of the image when you change the image. Images should be aggressively cached for performance reasons, and unless you change the name of an image, you risk clients getting a stale copy. Historically, after an image was edited, the name had to be changed and each reference to the image in the web app needed to be updated. Not only is this very labor intensive, it's also error prone (you could miss a reference, accidentally enter the wrong string, etc.) The built-in `ImageTagHelper <https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs>`__ can do this for you automatically. The ``ImageTagHelper`` can append a version number to the image name, so whenever the image changes, the server automatically generates a new unique version for the image. Clients are guaranteed to get the current image. This robustness and labor savings comes essentially free by using the ``ImageTagHelper``.  
+
+Most of the built-in Tag Helpers target existing HTML elements and provide server-side attributes for the element. For example, the ``<input>`` element used in many of the views in the *Views/Account* folder contains the ``asp-for`` attribute, which extracts the name of the specified model property into the rendered HTML. The following Razor markup:
+
+.. code-block:: html
+
+   <label asp-for="Email"></label>
+
+Generates the following HTML:
+
+.. code-block:: html
+
+   <label for="Email">Email</label>
+   
+The ``asp-for`` attribute is made available by the ``aspFor`` property in the ``LabelTagHelper``. See :doc:`authoring` for more information.
```

Comment:
aspFor => For

We [change](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.TagHelpers/LabelTagHelper.cs#L47) the HTML name with `[HtmlAttributeName]`.